texastribune/donations
# Update membership page with admin tools

#### What's this PR do?
Adds a new look and new features to the membership page for admin users.

#### Why are we doing this? How does it help us?
These features allow the admin team to cancel memberships and update cc info from the portal. These features will also be rolled out to members themselves in early September.

#### How should this be manually tested?
* Navigate to [UMP on staging](https://donations-testing.herokuapp.com/account/)
    * If you have an issue logging in, let me know
* In the form at the bottom left corner of the page, enter the email address "matthew.mccrary@texastribune.org" in the search field and click "View As"
    * If you don't see this form, let me know
    * I'm purposefully being vague here and leaving out screenshots, so message me off-site for any questions
* Click on the membership link in the left nav to get to the membership page
    * <img width="479" alt="Screen Shot 2023-08-21 at 8 42 45 AM" src="https://github.com/texastribune/donations/assets/88053677/f58ea4cb-0759-4186-b050-347ef640ac1a">
* Here you'll see a listing of all of the given user's recurring donations
    * <img width="1245" alt="Screen Shot 2023-08-21 at 8 44 33 AM" src="https://github.com/texastribune/donations/assets/88053677/4b3c15a6-9bec-4648-a083-e5d151bd4a94">
* With any recurring donation, click the "Edit" button in the "Payment Method" column
    * <img width="152" alt="Screen Shot 2023-08-18 at 2 39 54 PM" src="https://github.com/texastribune/donations/assets/88053677/9d2e7a8e-54cd-4774-b12f-3c5089df1eb7">
* A modal will pop up prompting you to update the card associated with the recurring donation. Use one of the cards listed [here](https://stripe.com/docs/testing#cards) (each card has instructions for CSC and expiration date, fyi) and click "Save"
    * <img width="489" alt="Screen Shot 2023-08-18 at 2 43 11 PM" src="https://github.com/texastribune/donations/assets/88053677/f75a14a7-22ab-4b17-9f99-d9ba146b9be7">
* After submittal, you should see a success message at the top of the page.
    * <img width="801" alt="Screen Shot 2023-08-21 at 9 22 13 AM" src="https://github.com/texastribune/donations/assets/88053677/c593e6d5-c724-4492-a6de-14e02c054b2b">
* Back at the recurring donation you updated, you should see that the info in the "Payment Method" column has updated to the new card info without having to do a refresh
    * <img width="699" alt="Screen Shot 2023-08-21 at 9 29 48 AM" src="https://github.com/texastribune/donations/assets/88053677/07b663b5-9e66-4c66-996a-976454edd747">
* Next, with any recurring donation, click the "Cancel" button in the "Next Payment" column
    * <img width="168" alt="Screen Shot 2023-08-21 at 9 27 32 AM" src="https://github.com/texastribune/donations/assets/88053677/837d284a-e7fd-4c0e-bf32-17e8ca3aaf4d">
* A confirmation modal will pop up. Click the button labeled "Yes".
    * <img width="719" alt="Screen Shot 2023-08-21 at 9 29 55 AM" src="https://github.com/texastribune/donations/assets/88053677/cf84d626-73d2-41ce-a173-869f9d3907ee">
* A success message will appear at the top of the page.
    * <img width="765" alt="Screen Shot 2023-08-21 at 9 32 10 AM" src="https://github.com/texastribune/donations/assets/88053677/83ad4b80-8c05-48e3-95b8-cbe4e9520b38">
* Looking through the table, the recurring donation you just cancelled will be removed, without having done a page reset.
    * <img width="802" alt="Screen Shot 2023-08-21 at 9 32 01 AM" src="https://github.com/texastribune/donations/assets/88053677/e77d267b-7f07-427f-afef-ba37bce081f8">

#### How should this change be communicated to end users?
We'll work closely with Kassie and Morgan on this one, so they'll be in the know.

#### Are there any smells or added technical debt to note?
Only that some old components (i.e. CardUpdate, etc.) will linger until we introduce these features to members as well. At that point, we can remove the legacy pieces that are no longer in use.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recj1ZV3meUBlHQJs?blocks=hide
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recRufDMoJ3pzcsvA?blocks=hide
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recVWcAMbah0TjMW9?blocks=hide
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recnM3mEWMfaxFBBv?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
